pkg/store/gcworker/gc_worker.go

```diff
@@ -1209,89 +1158,6 @@ func (w *GCWorker) legacyResolveLocks(
 	return nil
 }
 
-// resolveLocksPhysical uses TiKV's `PhysicalScanLock` to scan stale locks in the cluster and resolve them. It tries to
-// ensure no lock whose ts <= safePoint is left.
-func (w *GCWorker) resolveLocksPhysical(ctx context.Context, safePoint uint64) error {
-	metrics.GCWorkerCounter.WithLabelValues("resolve_locks_physical").Inc()
-	logutil.Logger(ctx).Info("start resolve locks with physical scan locks", zap.String("category", "gc worker"),
-		zap.String("uuid", w.uuid),
-		zap.Uint64("safePoint", safePoint))
-	startTime := time.Now()
-
-	registeredStores := make(map[uint64]*metapb.Store)
-	defer w.removeLockObservers(ctx, safePoint, registeredStores)
```

Comment:
The lock observer related functions are also part of Green GC which can be removed. So is the function `physicalScanAndResolveLocks`. Please check it again and clean all of them up.